### PR TITLE
new file

### PR DESCRIPTION
**Pull request by suchthis**
none

### CODE REVIEW

**Review by suchthis — COMMENTED** [review]
**Principle goals (May 4 - July 23):**

1. Convert data from different sources (PageFreezer, Versionista) into a common form and storing them in one place. If it is already handled, understand the database schema and the data stored in it.

2. Improve the method of computing differences. This will consist of two sub-tasks:

	2a. The changes can be categorized based on the type of object that has changed (text, image, hyperlinks etc.). The first sub-task will be to separate the changes into categories.

	2b. The second sub-task will be to extract meaningful features from the data of each type. This will include extracting information like type and length of the changed matter in a given text. If required, complex techniques can be used to extract features (NLP, Image Processing).

3. Provide a list of prioritized changes to analysts for reviewing and adding annotations.


**Stretch goals (July 23 - August 29):**

1. Write scripts for different Machine Learning Classification Algorithms which will learn to prioritize the changes. Each script will consist of four basic parts:

	* Data loading and pre-processing (if required) to convert data to input format of algorithm. Also, division of data into training and validation sets.
	* Machine Learning model creation using libraries like numpy, scikit-learn, Tensorflow etc.
	* Evaluation of model performance (using evaluation metrics) and creation of visualizations from the results of the validation set.
	* Storing the trained model by serializing the model with the help of modules like pickle, joblib.

2. Retrieve the trained models by deserializing the stored files and using them to prioritize new changes. Enhance the process of providing a list of prioritized changes to analysts for reviewing and adding annotations. An important step here will be to add correctly predicted cases to the training dataset. The model will be re-trained only after a sufficient amount of correctly predicted data is added.
